modules/client/src/interfaces.ts

```diff
@@ -524,3 +529,44 @@ export type ContractPermissionWithConditionParams = [
   string,
 ];
 export type ContractWithdrawParams = [string, string, BigNumber, string];
+
+export enum PluginSortBy {
+  SUBDOMAIN = "subdomain",
+}
+
+export type PluginQueryParams = Pagination & {
+  sortBy?: PluginSortBy;
+  subdomain?: string;
+};
+
+export type SubgraphPluginRepoReleaseListItem = {
+  release: number;
+  metadata: string;
+  builds: {
+    build: number;
+  }[];
+};
+
+export type SubgraphPluginRepoListItem = {
+  id: string;
+  subdomain: string;
+  releases: SubgraphPluginRepoReleaseListItem[];
+};
+
+export type PluginRepoReleaseMetadata = {
+  name: string;
+  description: string;
+  images: Object; // TODO specify parameters
+};
+
+export type PluginRepoRelease = {
+  release: number;
+  metadata: PluginRepoReleaseMetadata;
+  builds: number[];
```

Comment:
why `number[]`? Can't we add any other data here?
Having an array like `[1,2, 3, 4, 5, 6, 7]` would seem a bit redundant

Reply:
There is not much to add, a release can have multiple builds, but all the fields are common

Reply:
But every build has its own metadata as well. I would personally change this to a field like `lastBuildNumber: 7` and then add a separate functio like `getPlugin()` which did the fetching of all the metadata's (releases and builds)

Reply:
Mmm, i didnt know that ecah build has different metadata, I though it was shared between bulds. That messes things a bit, but we can follow the strategy of having a shorter response for `getPlugins` and the full response for `getPlugin`